### ISSUE TITLE
got.stream() doesn't return full response.

### ISSUE THREAD

**Issue by just-boris** [sindresorhus/got#152]
I am trying to use got for downloads and want to have a progress bar for it. Here is a gist how I am doing it: https://gist.github.com/just-boris/20d6f97795a115d3f548

I found out that I am getting not all content. The issue occures when I have an extra listner to the `response` object:

```js
got.stream(...).on('response', function(response) {
  res.on('data', function (chunk) {}) // if skip this, it will work
}).pipe(fs.createWriteStream(...))
```

Then I am getting incomplete result in the output file.

--
Found on `got@5.3.0` with latest `6.0.1` it works fine, but I can't use it, since I have to support earlier Node versions.

**Comment by floatdrop** [comment]
Hm... Maybe `6.0.0` works, because of switch to `duplexer2` from `duplexify` - but I'm not sure. Meanwhile you can look how [download](http://npmjs.com/download) and [download-status](http://npmjs.com/download-status) solved this problem.

---

Related #148

**Comment by just-boris** [comment]
Now it work as expected, thanks!

**Comment by just-boris** [comment]
Do you make a release for 5.x branch?

**Comment by floatdrop** [comment]
@just-boris yes, released as `5.3.1`.
